store/Auth/slice.ts

```diff
@@ -0,0 +1,183 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../../types/types';
+import { supabase } from '../../utils/supabase';
+import { RootState } from '../store';
+
+interface AuthState {
+  currentUser?: User;
+  loading: boolean;
+  error?: string;
+}
+
+const initialState: AuthState = {
+  currentUser: undefined,
+  loading: false,
+  error: undefined,
+};
+
```

Comment:
Add a type for **CredentialsPayload** to use in place of { username: string; password: string } _(see suggestions below for where)_
```suggestion
type CredentialsPayload = { username: string; password: string };
```